doc/devel/contribute.rst

```diff
@@ -279,11 +231,75 @@ though not necessarily all at the same time:
 - It involves Python features such as decorators and context managers, which
   have subtleties due to our implementation decisions.
 
+.. _first_contribution:
+
+First Contributions
+-------------------
+
+If this is your first open source contribution, or first time contributing to Matplotlib,
+and you need help or guidance finding a good first issue, look no further. This section will
+guide you through each step.
+
+1. Navigate to the `issues page <https://github.com/matplotlib/matplotlib/issues/>`_.
+2. Filter labels with `"Difficulty: Easy" <https://github.com/matplotlib/matplotlib/labels/Difficulty%3A%20Easy>`_ & `"Good first Issue" <https://github.com/matplotlib/matplotlib/labels/good%20first%20issue>`_ (optional).
+3. Click on an issue you would like to work on, and check to see if the issue a PR has been opened to resolve the issue.
```

Comment:
The three I can think of right now are: 

1. Add an extra step here i.e. "A good way to judge if you chose a suitable issue is by asking yourself 'Can I independently submit a PR in 1-2 weeks?'" This wording still fosters collaboration, and is telling the person to self check. 
2. Add a comment within the [PULL_REQUEST_TEMPLATE.md](https://github.com/matplotlib/matplotlib/blob/main/.github/PULL_REQUEST_TEMPLATE.md?plain=1) to say something along the lines of "Please explain in 1-3 sentences your implementation choices." This should be added into the summary section as well as a new check in the checklist. I would just consider how this would affect the larger contributor population. Personally, this could be good for the admins because their rationales would be explicit, so you'd save some time trying to sort out their logic/thought process. 
3. Combination of 1 & 2: add the additional text requirement in the md file, and add an additional step between steps 3 & 4 in `First Contributions' (see below). Optional (recommended): pull in the checklist within the [PULL_REQUEST_TEMPLATE.md](https://github.com/matplotlib/matplotlib/blob/main/.github/PULL_REQUEST_TEMPLATE.md?plain=1)

I know this creates redundant content, but for the sake of minimizing friction for the contributor, I think step 3 is the best route while adding the checklist to the contribution.rst file.

Reply:
Awesome, do you want me to make changes to the pull_request_template.md to include the implementation choices part in this pr as well?